sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -9,20 +9,27 @@
 using System.Reflection;
 using NUnit.Framework;
 using NUnit.Framework.Interfaces;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.ResourceManager.Resources;
 
 namespace Azure.Core.TestFramework
 {
     [Category("Recorded")]
-    public abstract class RecordedTestBase : ClientTestBase
+    public abstract class RecordedTestBase<TEnvironment> : ClientTestBase
+    where TEnvironment : TestEnvironment, new()
     {
         protected RecordedTestSanitizer Sanitizer { get; set; }
 
         protected RecordMatcher Matcher { get; set; }
 
         public TestRecording Recording { get; private set; }
 
-        public RecordedTestMode Mode { get; set; }
+        private static TimeSpan ZeroPollingInterval { get; } = TimeSpan.FromSeconds(0);
```

Comment:
This has a single call site I think we can inline the usage.

Reply:
Do we want to make these type of changes now? I just wanted to this stuff moved over from common really.

Reply:
This won't be a large change and would decrease the diff size.

Reply:
We are staying with parity with this PR